projects/media-viewer/src/lib/store/store-utils.ts

```diff
@@ -26,6 +27,52 @@ export class StoreUtils {
       }, {});
   }
 
+  static genTagNameEntities(annotations) {
+    const filterAnnoWithNoCommentsTags = annotations.filter(a => (a.comments.length && a.tags.length));
+    const allTags = filterAnnoWithNoCommentsTags.map(anno => this.groupByKeyEntities(anno.tags, 'name'));
+    const groupedByName = allTags.reduce(
+      (tagEntitiy: { [id: string]: Annotation }, tagItem) => {
+        return {
+          ...tagEntitiy,
+          ...tagItem
+        };
+      }, {});
+
+    return this.genNameEnt(annotations, groupedByName)
+  };
+
+  static genNameEnt(annos, groupedByName) {
+     return Object.keys(groupedByName).reduce(
+      (tagNameEnt, key) => {
+        const annot = annos.map(anno => {
+          const tags = anno.tags.map(tag => {
+            if (tag.name === key) {
+              return anno.id;
+            }
+          });
+          return tags.filter(a => a !== undefined);
+        });
```

Comment:
a bit confused by the logic here, but isn't it possible to do it like this: 
        annos.filter(anno => anno.tags.find(tag => tag.name === name))
          .map(anno => anno.id)